pandas/tests/groupby/test_groupby.py

```diff
@@ -2521,6 +2521,34 @@ def test_groupby_cumprod(self):
         expected.name = 'value'
         tm.assert_series_equal(actual, expected)
 
+    def test_groupby_cum_skipna(self):
+        # GH 19806
+        # make sure that skipna works for both cumsum and cumprod
+        df = pd.DataFrame({'key': ['b'] * 10 + ['a'] * 2, 'value': 3})
+        df.at[3] = np.nan
+        df.at[3, 'key'] = 'b'
+
```

Comment:
ideally you could parameterize this over skipna=True/False

Reply:
I parametrized it over cumprod/sum & skipna.